pkg/app/ops/orphancommandcleaner/orphancommandcleaner.go

```diff
@@ -0,0 +1,85 @@
+package orphancommandcleaner
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/pipe-cd/pipe/pkg/datastore"
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+var (
+	commandTimeOut              = 24 * time.Hour
+	interval                    = 6 * time.Hour
+	maxConsecutiveFailuresCount = 3
+)
+
+type OrphanCommandCleaner struct {
+	commandstore datastore.CommandStore
+	logger       *zap.Logger
+}
+
+func NewOrphanCommandCleaner(
+	ds datastore.DataStore,
+	logger *zap.Logger,
+) *OrphanCommandCleaner {
+	return &OrphanCommandCleaner{
+		commandstore: datastore.NewCommandStore(ds),
+		logger:       logger.Named("orphan-command-cleaner"),
+	}
+}
+
+func (c *OrphanCommandCleaner) Run(ctx context.Context) error {
+	consecutiveFailuresCount := 0
+
+	t := time.NewTicker(interval)
+	for {
+		select {
```

Comment:
Please check the `context.Done` to exit when `ops` was terminated.

``` golang
for {
    select {
          case <- ctx.Done():
                 return nil
         case <- t.C:
             ...
}
}
```

Reply:
Oops, thanks